app/api/v1/common/type_mapper.py

```diff
@@ -13,26 +13,30 @@ class TypeMapper:
     def string_to_element_mod_p(value: str) -> ElementModP:
         elementmodp = hex_to_p(value)
         if elementmodp is None:
-            raise ValueError("invalid key: " + value)
+            raise ValueError(TypeMapper.type_error_message(value, "element_mod_q"))
         return elementmodp
 
     @staticmethod
     def string_to_element_mod_q(value: str) -> ElementModQ:
         elementmodq = hex_to_q(value)
         if elementmodq is None:
-            raise ValueError("invalid key: " + value)
+            raise ValueError(TypeMapper.type_error_message(value, "element_mod_q"))
         return elementmodq
 
     @staticmethod
     def int_to_element_mod_p(value: int) -> ElementModP:
         elementmodp = int_to_p(value)
         if elementmodp is None:
-            raise ValueError("invalid key: " + str(value))
+            raise ValueError(TypeMapper.type_error_message(str(value), "element_mod_q"))
         return elementmodp
 
     @staticmethod
     def int_to_element_mod_q(value: int) -> ElementModQ:
         elementmodq = int_to_q(value)
         if elementmodq is None:
-            raise ValueError("invalid key: " + str(value))
+            raise ValueError(TypeMapper.type_error_message(str(value), "element_mod_q"))
         return elementmodq
+
+    @staticmethod
```

Comment:
Doesn't have to be in this class. I would have made just a stand alone function. Fairly common in python and results in less static methods.

Reply:
Having a separate class with a static like this is a common practice in C# because you have to do it that way.  This means only a single class has to have the static methods and can be used by other files by importing just the class and not each function separately.  We should have a common pattern that is used for the projects for this type of thing.

Reply:
Re: avoid static: good point, also static isn't mockable, so it isn't unit testing friendly.  Although I can't tell yet if that's particularly important for this repo and language.  Regardless, I'll convert it to stand-alone functions if that's idiomatic for Python.

Re: establish a pattern for this situation: agreed, I like consistency.  However, I'm not comfortable enough with the codebase or language yet to have an opinion on what patterns should be or when to apply them.

Re: additional imports.  It's only one additional import in this instance, so I'm inclined to take Keith's advice, convert to functions, and not worry about implications of this decision and patterns for the future just yet.